src/client/InteractionClient.js

```diff
@@ -0,0 +1,207 @@
+'use strict';
+
+const BaseClient = require('./BaseClient');
+const ApplicationCommand = require('../structures/ApplicationCommand');
+const Interaction = require('../structures/Interaction');
+const { Events, ApplicationCommandOptionType, InteractionType, InteractionResponseType } = require('../util/Constants');
+
+let sodium;
+
+/**
+ * Interaction client is used for interactions.
+ *
+ * @example
+ * const client = new InteractionClient({
+ *   token: ABC,
+ *   publicKey: XYZ,
+ * });
+ *
+ * client.on('interactionCreate', () => {
+ *   // automatically handles long responses
+ *   if (will take a long time) {
+ *     doSomethingLong.then((d) => {
+ *       interaction.reply({
+ *         content: 'wow that took long',
+ *       });
+ *     });
+ *   } else {
+ *     interaction.reply('hi!');
+ *   }
+ * });
+ * ```
+ */
+class InteractionClient extends BaseClient {
+  /**
+   * @param {Options} options Options for the client.
+   * @param {undefined} client For internal use.
+   */
+  constructor(options, client) {
+    super(options);
+
+    Object.defineProperty(this, 'token', {
+      value: options.token,
+      writable: true,
+    });
+
+    Object.defineProperty(this, 'clientID', {
+      value: options.clientID,
+      writable: true,
+    });
+
+    Object.defineProperty(this, 'publicKey', {
+      value: options.publicKey ? Buffer.from(options.publicKey, 'hex') : undefined,
+      writable: true,
+    });
+
+    // Compat for direct usage
+    this.client = client || this;
+    this.interactionClient = this;
+  }
+
+  /**
+   * Get registered slash commands.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {Command[]}
+   */
+  async getCommands(guildID) {
```

Comment:
```suggestion
   * Fetch registered slash commands.
   * @param {Snowflake} [guildID] Optional guild ID.
   * @returns {Command[]}
   */
  async fetchCommands(guildID) {
```
`fetch` might make more sense to keep consistency with all other methods that do an API request.